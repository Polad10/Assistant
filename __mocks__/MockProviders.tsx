import { Dispatch, PropsWithChildren, SetStateAction } from 'react'
import { Translator } from '../helpers/Translator'
import { AppointmentRequest } from '../modals/Appointment'
import { PatientRequest } from '../modals/Patient'
import { Setting } from '../modals/Setting'
import { TreatmentRequest } from '../modals/Treatment'
import { PaymentRequest } from '../modals/Payment'
import { Api } from '../helpers/Api'
import { mockModels } from './MockModels'
import { AuthContext } from '../contexts/AuthContext'
import { DataContext } from '../contexts/DataContext'
import { ThemeContext } from '../contexts/ThemeContext'
import { LocalizationContext } from '../contexts/LocalizationContext'

const themeContextMock = {
  primary: '#121212',
  secondary: '#1a1a1a',
  accent: '#7600eb',
  neutral: '#faf8ff',
  border: '#374151',
  info: '#d3fbd8',
  success: '#00e5b0',
  warning: '#9f1239',
  error: '#9f1239',
}

const localizationContextMock = {
  translator: new Translator('en'),
  language: 'en',
}

const authContextMock = {
  user: undefined,
  loading: false,
}

const dataContextMock = {
  fetchPatients: async () => {},
  fetchAppointments: async () => {},
  fetchTreatments: async () => {},
  fetchPayments: async () => {},
  fetchSetting: async () => {},

  createAppointment: async (appointmentRequest: AppointmentRequest) => mockModels.appointment_1,
  updateAppointment: async (appointmentRequest: AppointmentRequest) => mockModels.appointment_1,
  deleteAppointment: async (appointmentId: number) => {},

  createPatient: async (patientRequest: PatientRequest) => mockModels.patient_1,
  updatePatient: async (patientRequest: PatientRequest) => mockModels.patient_1,
  deletePatient: async (patientId: number) => {},

  createTreatment: async (treatmentRequest: TreatmentRequest) => mockModels.treatment_1,
  updateTreatment: async (treatmentRequest: TreatmentRequest) => mockModels.treatment_1,
  deleteTreatment: async (treatmentId: number) => {},

  createPayment: async (paymentRequest: PaymentRequest) => mockModels.payment_1,
  updatePayment: async (paymentRequest: PaymentRequest) => mockModels.payment_1,
  deletePayment: async (paymentId: number) => {},

  createSetting: async (setting: Setting) => {},
  updateSetting: async (setting: Setting) => {},

  api: undefined,
  setApi: (action: SetStateAction<Api | undefined>): Dispatch<SetStateAction<Api | undefined>> => jest.fn(),

  loading: false,
  setLoading: (action: SetStateAction<boolean>): Dispatch<SetStateAction<boolean>> => jest.fn(),

  patients: [mockModels.patient_1, mockModels.patient_2],
  appointments: [mockModels.appointment_1, mockModels.appointment_2],
  treatments: [mockModels.treatment_1, mockModels.treatment_2],
  payments: [mockModels.payment_1, mockModels.payment_2],
  setting: mockModels.setting,
}

export default function MockProviders({ children }: PropsWithChildren) {
  return (
    <AuthContext.Provider value={authContextMock}>
      <DataContext.Provider value={dataContextMock}>
        <ThemeContext.Provider value={themeContextMock}>
          <LocalizationContext.Provider value={localizationContextMock}>{children}</LocalizationContext.Provider>
        </ThemeContext.Provider>
      </DataContext.Provider>
    </AuthContext.Provider>
  )
}
