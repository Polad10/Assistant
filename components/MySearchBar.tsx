import { PropsWithChildren, SetStateAction, forwardRef, useContext, useImperativeHandle, useRef, useState } from 'react'
import { DeviceEventEmitter, StyleSheet, TextInput } from 'react-native'
import { SearchBar } from '@rneui/themed'
import { SearchBar as BaseSearchBar } from '@rneui/base'
import { Ionicons } from '@expo/vector-icons'
import { MaterialIcons } from '@expo/vector-icons'
import { ThemeContext, ThemeContextType } from '../contexts/ThemeContext'
import { translate } from '../helpers/Translator'

type Props = {
  placeholder: string
  searchEventName: string
}

type SearchBarRefType = {
  clear: () => void
}

const MySearchBar = forwardRef((props: Props, ref) => {
  const themeContext = useContext(ThemeContext)!
  const [search, setSearch] = useState('')

  const searchBarRef = useRef<TextInput & PropsWithChildren<BaseSearchBar>>(null)

  useImperativeHandle(ref, () => {
    return {
      clear,
    }
  })

  function clear() {
    if (searchBarRef.current) {
      searchBarRef.current.clear()
    }
  }

  function updateSearch(value: SetStateAction<string>) {
    setSearch(value)

    DeviceEventEmitter.emit(props.searchEventName, value)
  }

  return (
    <SearchBar
      ref={searchBarRef}
      placeholder={props.placeholder}
      platform={'ios'}
      containerStyle={styles(themeContext).searchBarContainer}
      inputContainerStyle={styles(themeContext).searchBarInputContainer}
      inputStyle={styles(themeContext).searchBarInput}
      searchIcon={<Ionicons name='search-outline' size={20} color={themeContext.neutral} />}
      clearIcon={<MaterialIcons name='clear' size={20} color={themeContext.neutral} onPress={clear} />}
      cancelButtonTitle={translate('cancel')}
      value={search}
      onChangeText={updateSearch}
    />
  )
})

const styles = (themeContext: ThemeContextType) =>
  StyleSheet.create({
    searchBarContainer: {
      backgroundColor: themeContext.primary,
    },
    searchBarInputContainer: {
      backgroundColor: themeContext.primary,
    },
    searchBarInput: {
      color: themeContext.neutral,
    },
  })

export default MySearchBar
export { SearchBarRefType }
