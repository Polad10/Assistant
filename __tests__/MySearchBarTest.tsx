import renderer from 'react-test-renderer'
import MockProviders from '../__mocks__/MockProviders'
import MySearchBar from '../components/MySearchBar'

describe('MySearchBar', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <MockProviders>
          <MySearchBar placeholder='placeholder' searchEventName='searchEventName' />
        </MockProviders>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})
