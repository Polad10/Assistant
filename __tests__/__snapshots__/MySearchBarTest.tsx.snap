// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MySearchBar renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#121212",
      "flexDirection": "row",
      "overflow": "hidden",
      "paddingBottom": 13,
      "paddingTop": 13,
    }
  }
  testID="RNE__SearchBar-wrapper"
>
  <View
    style={
      {
        "paddingHorizontal": 0,
        "width": "100%",
      }
    }
    testID="RNE__Input__view-wrapper"
  >
    <View
      collapsable={false}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#121212",
          "borderBottomWidth": 0,
          "borderColor": "#86939e",
          "borderRadius": 9,
          "flexDirection": "row",
          "marginLeft": 8,
          "marginRight": 8,
          "minHeight": 36,
          "transform": [
            {
              "translateX": 0,
            },
          ],
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "height": 40,
            "justifyContent": "center",
            "marginLeft": 8,
            "marginVertical": 4,
            "paddingRight": 4,
          }
        }
      >
        <Text />
      </View>
      <TextInput
        editable={true}
        onBlur={[Function]}
        onCancel={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        placeholder="placeholder"
        placeholderTextColor="#7d7d7d"
        platform="ios"
        style={
          {
            "color": "#faf8ff",
            "flex": 1,
            "fontSize": 18,
            "marginLeft": 6,
            "minHeight": 40,
            "overflow": "hidden",
          }
        }
        testID="RNE__SearchBar"
        underlineColorAndroid="transparent"
        value=""
      />
      <View
        style={
          {
            "alignItems": "center",
            "height": 40,
            "justifyContent": "center",
            "marginRight": 8,
            "marginVertical": 4,
            "paddingRight": 4,
          }
        }
      >
        <View
          style={
            {
              "flexDirection": "row",
            }
          }
        />
      </View>
    </View>
    <Text
      style={
        {
          "color": "#ff190c",
          "fontSize": 12,
          "height": 0,
          "margin": 0,
          "padding": 0,
        }
      }
    />
  </View>
  <View
    onLayout={[Function]}
    style={
      {
        "opacity": 0,
        "position": "absolute",
        "right": null,
      }
    }
    testID="RNE__SearchBar-cancelButtonContainer"
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
    >
      <View
        style={{}}
        testID="RNE__SearchBar-cancelButton"
      >
        <Text
          style={
            {
              "color": "#007aff",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            }
          }
        >
          Cancel
        </Text>
      </View>
    </View>
  </View>
</View>
`;
